src/uct/base/uct_iface.h

```diff
@@ -502,7 +502,7 @@ typedef struct {
         _req = ucs_container_of(_priv, uct_pending_req_t, priv); \
         ucs_queue_pull_non_empty(_queue); \
         _status = _req->func(_req); \
-        if (_status != UCS_OK) { \
+        if ((_status == UCS_ERR_NO_RESOURCE) || (_status == UCS_INPROGRESS)) { \
```

Comment:
why needed, where is any other error returned?

Reply:
yes, in case of TCP, it could be CONN_RESET

Reply:
where from exactly? it should return UCS_OK if got CONN_RESET

Reply:
then, need to fix ucp_wireup_msg_progress to return UCS_OK when got an error, to remove from pending queue

Reply:
not only `ucp_wireup_msg_progress()` has to be fixed then:
e.g. `ucp_tag_eager_zcopy_single`->`ucp_do_am_zcopy_single` has to be fixed as well

Reply:
ic, then need to fix UCP, i guess before this change UCT transports did not really return error from send functions

Reply:
other TLs that don't use this macro (i.e. all, expect TCP), they arbiter dispatch function is responsible to REMOVE_ELEM in case of error
but this macro used by TCP, since TCP doesn't use arbiter

Reply:
other transports (e.g RC) don't have different handling for UCS_ERR_NO_RESOURCE and other error

Reply:
> other transports (e.g RC) don't have different handling for UCS_ERR_NO_RESOURCE and other error

yes, you're right
will fix all places in UCP then...

Reply:
will fix in other UCT TLs to handle error in dispatch function